Sensor: VL53L1X: Add VL53L1X driver

pablogs9 opened the pull request:
Add support for ST VL53L1X TOF sensor.

Signed-off-by: Pablo Garrido pablogs9@gmail.com

erwango commented on drivers/sensor/vl53l1x/vl53l1_platform_user_config.h:
```diff
@@ -1,29 +1,8 @@
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions are met:
-	* Redistributions of source code must retain the above copyright
-	  notice, this list of conditions and the following disclaimer.
```
Deleting these lines is a clear violation of the copyright.

pablogs9 replied:
Sorry, I'm going to check the references PR and make change in order see if its possible to integrate this driver. Thanks.

erwango reviewed (changes requested):
Thanks for this contribution!

Please note that there was an initial effort to do the same change (https://github.com/zephyrproject-rtos/zephyr/pull/18772). You can go through it as it might includes interesting hints.

Before going into the details of the works, I've noted 2 things to be sorted out first:

Files origin: you indicate that you got the files from a third party site. Please get the files from the official ST website (https://www.st.com/en/embedded-software/stsw-img007.html). Aim is to ensure we're starting from a known version (last 2.3.3) compatible with the version available on https://github.com/zephyrproject-rtos/hal_st.

Files copyright and ownership:
Files distributed by ST are available under BDS-3 license  with ST Copyright. Following discussions in #18772 ST allows to change License for these files to Apache 2.0 to fir Zephyr policy, so you're allowed to upload directly these files under the Apache 2.0 License (a mention in the commit message to mention that would be nice).
Though, ST copyright should be retained. You are not allowed to replace ST copyright by yours. In case of consequent changes of the files (let's say this does not include style changes), you have the possibility to *add* your own copyright.

pablogs9 commented:
@erwango Please check if copyright and ownership its ok for you now? Sorry about the inconvenience.

pablogs9 commented:
@erwango please, could you tell me the license format for `vl53l1_platform_user_config.h` in order to pass the Scancode check?

erwango commented on drivers/sensor/vl53l1x/vl53l1_platform_user_config.h:
```diff
@@ -0,0 +1,84 @@
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
You can remove this now. My point was to sensitize you of this side of things. All things are good now.
You are authorized to delete it.

pablogs9 replied:
Done, thanks

erwango reviewed (changes requested):
Last comment on the License aspect.
I'll take more time to review the technical side

erwango reviewed (changes requested):
Sorry for delay in reviewing this.
Please split your change in (at least) 2 commits
- First commit with st lib unmodifed files (in the commit message, explain where you get the files from, the version and dependency with hal_st module
- Second commit with the changes you introduce for the adaptation.

pablogs9 committed:
Sensor: VL53L1X: Added VL53lL1 Platform files

Version 2.3.3
Downloaded from: https://www.st.com/content/st_com/en/products/embedded-software/proximity-sensors-software/stsw-img007.html?dl=13IzPtVN2sSiD%2FNb7m%2BJJg%3D%3D%2CZ3i%2BFXm12juv4gHEDxyIahfNjkAyaumiWlNlgIqumXmv%2F5wxmJgPjLudn3mueSxfSJ%2FZzUKLokPO8MeEBbrjWDZSZZNi7UP0waDRi3k8j%2FATrYqzlslfNjuhrvPCDGkXtf71CP8ozQ2SYk0yDRYZcKhEL1bMgrRYF3%2B0LkexqyBgU3bq8rLD82x4E2Ipvc%2B5mEJkSrfpX5kwNi%2B7VUAARQJASBIyjNb5G4bLIoFHvwBzceQAj9fv0nSbuPOQuQ1PkYb82uNOCgWiSrCxgrBoahNcmdz2C%2FO4cWmgnZTFdIboztMHoiXCHCimyPOiuTJF9InSaXpGqi%2BanHaUZuSGqg%3D%3D&uid=RRvtWpOX61pMwc+XREWMF16HjOvr6UsIf+4dF+vE1+Y=

Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

pablogs9 commented:
Hello, @erwango here you have the two commits, first of them is just the files downloaded from [here](https://www.st.com/content/st_com/en/products/embedded-software/proximity-sensors-software/stsw-img007.html?dl=13IzPtVN2sSiD%2FNb7m%2BJJg%3D%3D%2CZ3i%2BFXm12juv4gHEDxyIahfNjkAyaumiWlNlgIqumXmv%2F5wxmJgPjLudn3mueSxfSJ%2FZzUKLokPO8MeEBbrjWDZSZZNi7UP0waDRi3k8j%2FATrYqzlslfNjuhrvPCDGkXtf71CP8ozQ2SYk0yDRYZcKhEL1bMgrRYF3%2B0LkexqyBgU3bq8rLD82x4E2Ipvc%2B5mEJkSrfpX5kwNi%2B7VUAARQJASBIyjNb5G4bLIoFHvwBzceQAj9fv0nSbuPOQuQ1PkYb82uNOCgWiSrCxgrBoahNcmdz2C%2FO4cWmgnZTFdIboztMHoiXCHCimyPOiuTJF9InSaXpGqi%2BanHaUZuSGqg%3D%3D&uid=RRvtWpOX61pMwc+XREWMF16HjOvr6UsIf+4dF+vE1+Y=) v2.3.3.

The second is the modifications required and the modified licenses (as we agreed).

pablogs9 committed:
Sensor: VL53L1X: Add VL53L1X driver

Add support for ST VL53L1X TOF sensor.

Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

erwango commented on drivers/sensor/vl53l1x/vl53l1_platform_user_config.h:
```diff
@@ -0,0 +1,118 @@
+/*
```
This file is not used in case of vl53l0x, is it needed in this case ?

erwango commented on drivers/sensor/vl53l1x/vl53l1_platform_user_data.h:
```diff
@@ -0,0 +1,99 @@
+
```
This file is not used in case of vl53l0x, is it needed in this case ?

erwango commented on drivers/sensor/vl53l1x/vl53l1_platform_user_defines.h:
```diff
@@ -0,0 +1,128 @@
+/*
```
This file is not used in case of vl53l0x, is it needed in this case ?

erwango reviewed (changes requested):
Thanks for this update.

Some requests on the first commit:
- Please use directly the Apache 2.0 Licence boilerplate, as is you've been delivered it directly this way by ST (my bad if the request was not clear enough).
- Apply a dos2unix conversion on the files. Indeed, in the second commit, all the lines are modified. I guess this is your editor which apply this conversion (which is ok), but this is preventing from identifying easily what changes you have introduced in the files. So let's make it directly in the first commit. I'll review the second commit when we'll be able to see the changes you introduced (which I hope should happen if you apply these guidelines) 
- Please check if all the files are needed. Some of the files you pushed are not present in case of vl53l0x driver.
- Use the following commit message:
"
drivers/sensor: vl53l1x: Add platform file from vl53l1x ST library (v2.3.3)
 
URL:
       https://www.st.com/en/embedded-software/stsw-img007.html
 Version:
       2.3.3
 Maintained-by:
       External
 License:
       Apache 2.0 (With authorization from STMicroelectronics)
"

galak commented:
@pablogs9 any updates on the comments from @erwango ?
